pkg/base/capacityplanner/node_capacity.go

```diff
@@ -107,17 +107,15 @@ func (nc *nodeCapacity) selectACForVolume(vol *genV1.Volume) *accrd.AvailableCap
 	// try to find free capacity
 	var foundAC *accrd.AvailableCapacity
 	for _, acs := range filteredMap {
-		foundAC = searchACWithClosestSize(acs, size)
+		foundAC = searchACWithClosestSize(acs, size, nil)
 		if foundAC != nil {
 			break
 		}
 	}
 	// for LVG SC try to reserve AC to create new LVG since no free space found on existing
 	if isLVG && foundAC == nil {
-		// for the new lvg we need some extra space
-		size += LvgDefaultMetadataSize
 		// search AC in sub storage class
-		foundAC = searchACWithClosestSize(scToACMap[subSC], size)
+		foundAC = searchACWithClosestSize(scToACMap[subSC], size, SubtractLVMMetadataSize)
```

Comment:
It's really hard to understand the logic in `selectACForVolume`. Does it make sense to refactor and split code for LVG and non LVG case?

Reply:
I agree function is quite complex. I think refactoring can be nontrivial, so it will be better to do this as a separate activity.